Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -369,6 +377,30 @@ inline bool ShouldDraw() {
         return CurrentGState().ShouldDraw();
     }
 
+    inline void PushBeginDraw() {
+        if ((_beginEndDrawDepth)++ == 0) {
+            deviceContext->BeginDraw();
+        }
+    }
+
+    inline void PopEndDraw() {
+        if (--(_beginEndDrawDepth) == 0) {
+            FAIL_FAST_IF_FAILED(deviceContext->EndDraw());
```

Comment:
The _true_ recommendation was to have inner functions return `HRESULT` and FAIL_FAST or ignore the error gracefully at the C API boundary.